Update git_part1 + cac_mang_cuc_bo

thaonguyentien opened the pull request:
Update git la gi trong git+part
Viet ve cac mang cuc bo VLAN, GRE,VXLAN commit time 1.

thaonguyentien committed:
Update git_part1 + cac_mang_cuc_bo

Update git la gi trong git+part
Viet ve cac mang cuc bo VLAN, GRE,VXLAN commit time 1.

NTT-TNN commented:
@hieulq @haminhcong @vanduc95 @huynhducbk95  Review em với các anh.

haminhcong commented on K59/Nguyen-Tien-Thao/Git_tutorial/Git_Basic_Part1.md:
```diff
@@ -44,6 +44,8 @@ Trong các DVCS các máy khách không chỉ sao chép về máy cục bộ cá
+
 #### 2.1 Điểm khác biệt giữa Git và các VCS khác
 
 - Các hệ thống VCS khác mỗi khi người sử dụng commit VCS sẽ lưu lại toàn bộ các file. Còn với Git hệ thống sẽ lưu lại một **ảnh chụp(snapshot)** các file. Với những file đã thay đổi git sẽ tạo một phiên bản mới còn với những file không thay đổi git sẽ ánh xạ file đó vào file cũ đã không thay đổi.
```
Có một  cái ảnh lỗi em chưa sửa cho anh à ?

thaonguyentien replied:
@haminhcong  ảnh nào lỗi anh ơi?

haminhcong replied:
https://raw.githubusercontent.com/NTT-TNN/Basic_knowledge/master/images/File%20Status%20LifeCycle.PNG

haminhcong commented on K59/Nguyen-Tien-Thao/Git_tutorial/Git_Basic_Part1.md:
```diff
@@ -44,6 +44,8 @@ Trong các DVCS các máy khách không chỉ sao chép về máy cục bộ cá
+
 #### 2.1 Điểm khác biệt giữa Git và các VCS khác
 
 - Các hệ thống VCS khác mỗi khi người sử dụng commit VCS sẽ lưu lại toàn bộ các file. Còn với Git hệ thống sẽ lưu lại một **ảnh chụp(snapshot)** các file. Với những file đã thay đổi git sẽ tạo một phiên bản mới còn với những file không thay đổi git sẽ ánh xạ file đó vào file cũ đã không thay đổi.
```
"Các hệ thống VCS khác mỗi khi người sử dụng commit VCS sẽ lưu lại toàn bộ các file."

Anh nhớ là đoạn này anh đã nhắc em sửa ít nhất 1 lần rồi

haminhcong commented on K59/Nguyen-Tien-Thao/Git_tutorial/Git_Basic_Part1.md:
```diff
@@ -44,6 +44,8 @@ Trong các DVCS các máy khách không chỉ sao chép về máy cục bộ cá
 
 ### 2. Git
 
+Git là một VCS, cụ thể hơn là một DVCS sau đây ta sẽ đi vậy git khác với các VCS khác như thế nào, tại sao ta nên sử dụng git cũng như cách sử dụng git cơ bản.
```
"Git là một VCS, cụ thể hơn là một DVCS sau đây ta sẽ đi vậy git khác với các VCS khác như thế nào"

Em định nói cái gì đây ?

haminhcong commented on K59/Nguyen-Tien-Thao/Mang_May_Tinh/Tuan_2.Cac_mo_hinh_mang_cuc_bo.md:
```diff
@@ -63,10 +56,71 @@ Bây giờ nếu hai máy tính muốn gửi tin cho nhau chúng sẽ thực hi
+
+Trong hình giả sử máy A muốn gửi gói tin cho máy B gói tin từ máy A đến máy B sẽ thay đổi và được đóng gói và bóc tách cụ thể là:
+
+- Máy A tạo một Ethernet frame với địa chỉ nguồn là IP và MAC của A và địa chỉ đích là IP và MAC của B sau đó gửi tới VTEP-1.
```
L2 Frame không bao giờ chứa địa chỉ nguồn là IP. Đọc lại phần này

haminhcong replied:
Vẽ hình, lấy ví dụ như cách mà 2 tuần trước em hướng dẫn em.

thaonguyentien replied:
Em sẽ sửa sau anh à tối nay e bận làm bài tập lớn rồi anh @haminhcong

haminhcong commented on K59/Nguyen-Tien-Thao/Mang_May_Tinh/Tuan_2.Cac_mo_hinh_mang_cuc_bo.md:
```diff
@@ -63,10 +56,71 @@ Bây giờ nếu hai máy tính muốn gửi tin cho nhau chúng sẽ thực hi
+- Máy A tạo một Ethernet frame với địa chỉ nguồn là IP và MAC của A và địa chỉ đích là IP và MAC của B sau đó gửi tới VTEP-1.
+- VTEP-1 dựa vào mapping table để map giữa máy B và VTEP-2 và tiến hành đóng gói packet bằng cách thêm các trường: VXLAN,UDP and outer address header. Cụ thể trong outer address header IP nguồn và MAC nguồn là của VTEP-1 còn IP đích và MAC đích là của STEP-2. Sau đó VTEP-1 sử dụng địa chỉ IP của VTEP-2 để xác định nột tiếp theo sẽ chuyển tiếp.
+- Gói tin được chuyển tới VTEP-2. Sau khi tới VTEP-2 các trường outer Ethernet, IP, UDP, and VXLAN headers được loại bỏ và chuyển tới hostB
+
```
Lấy lại ví dụ cho anh. Anh muốn thấy khi dữ liệu đi qua các đối tượng trong hệ thống thì nó được đóng gói vào gói tin gì, khi nó đến các trạm cuối thì dữ liệu được bóc tách ra như thế nào. Viết từng bước 1 ra cho anh, sau mỗi bước dữ liệu có hình dạng như thế nào, bao gồm các trường gì, mỗi trường có giá trị là gì, em vẽ ra cho anh.

haminhcong commented on K59/Nguyen-Tien-Thao/Mang_May_Tinh/Tuan_2.Cac_mo_hinh_mang_cuc_bo.md:
```diff
@@ -1,12 +1,5 @@
-
 ## Flat
 
 Mạng flat là một mạng không cung cấp bất kỳ một tùy chọn thêm nào. Nó là mạng truyền thống được sử dụng ở Layer2. Mạng flat phân tách miền quảng bá bằng cách sử dụng router và mội máy kết nối đến cùng một router đều thuộc cùng một miền quản bá. Nói chung là tất cả các thiết bị trong một mạng flat có cùng một miền quảng bá.
```
"Mạng flat phân tách miền quảng bá bằng cách sử dụng router"

Xem lại câu này. Không đúng, vì nó mâu thuẫn với câu dưới:

 Nói chung là tất cả các thiết bị trong một mạng flat có cùng một miền quảng bá.

hieulq reviewed (changes requested):
Fix lại theo yêu cầu của Công.

thaonguyentien committed:
Update lai cac file

chinh sua anh gui goi tin toi facebook qua nat overload
chinh sua git la gi
ve va viet lai chuyen goi tin trong VXLAN

haminhcong commented on K59/Nguyen-Tien-Thao/Mang_May_Tinh/Tuan_2.Cac_mo_hinh_mang_cuc_bo.md:
```diff
@@ -63,10 +56,71 @@ Bây giờ nếu hai máy tính muốn gửi tin cho nhau chúng sẽ thực hi
+
+### VXLAN tunnel Endpoint(VTEP)
+
+VTEP là một thiết bị được VXLAN sử dụng để encapsulation and de-encapsulation các gói tin vận chuyển và ánh xạ các máy trong VXLAN. VTEP cung cấp hai interface một là switch interface trong mạng LAN hai là IP interface dùng trong IP network.
```
VTEP bao gồm những thành phần nào ? Và làm những nhiệm vụ nào ?

thaonguyentien replied:
em sẽ bổ sung ạ

haminhcong commented:
@huynhducbk95 @vanduc95 @HaManhDong vào review cho Thạo giúp mình với.

thaonguyentien committed:
Update

Chinh sua lai anh loi git_part1
Them hai thanh phan VTEP
Viet mot vai cau lenh ve Linux

vanduc95 commented on K59/Nguyen-Tien-Thao/Mang_May_Tinh/Tuan_2.Cac_mo_hinh_mang_cuc_bo.md:
```diff
@@ -63,10 +56,73 @@ Bây giờ nếu hai máy tính muốn gửi tin cho nhau chúng sẽ thực hi
+
+- Giới hạn 4094 VLANs là không đủ.
+- Hạn chế về khoảng cách và triển khai.
+- Bảng địa chỉ MAC của switch quá lớn.
```
Tại sao bảng địa chỉ MAC của switch lại quá lớn, nó khác gì bảng địa chỉ MAC khi không sử dụng VLAN?

thaonguyentien replied:
khác chứ anh. Khi không sử dụng VLAN (tức là không có mạng LAN ảo) thì switch sẽ chỉ cần lưu một số IP ứng với địa chỉ MAC nếu tím trong bảng đó không có thì sẻ gửi ARP request để tìm địa chỉ MAC đó còn với VLAN thì buộc phải lưu tất cả địa chỉ MAC để biết nó thuộc VLAN ID nào ạ.

vanduc95 replied:
Anh nghĩ em đang nhầm giữa `mac table` và `arp table` . Khi có hay không có VLAN thì `mac table` vẫn có một cấu trúc như vậy

Vlan    Mac Address       Type        Ports
----    -----------       --------    -----
   1    00ld.70ab.5d60    DYNAMIC     Fa0/2
   1    00le.f724.al60    DYNAMIC     Fa0/3

Mặc định khi không cấu hình VLAN thì tất cả các port trên switch đều thuộc VLAN 1, khi cấu hình VLAN thì nó chỉ thay trường **vlan** bằng VLANID của port đó. Vậy nên bảo "Bảng địa chỉ MAC của switch quá lớn" thì anh thấy hơi sai sai.
Kiểm tra lại nhé :))

thaonguyentien replied:
dạ vâng em sẽ check lại

vanduc95 commented on K59/Nguyen-Tien-Thao/Mang_May_Tinh/Tuan_2.Cac_mo_hinh_mang_cuc_bo.md:
```diff
@@ -63,10 +56,73 @@ Bây giờ nếu hai máy tính muốn gửi tin cho nhau chúng sẽ thực hi
 
 Giải pháp đó có một vấn đề nếu một switch có nhiều mạng VLAN kết nối sẽ tốn nhiều cổng cho và sẽ không hiệu quả. Một giải pháp khác là sử dụng một cổng đặc biệt trên mỗi switch gọi là trunk port để kết nối giữa hai switch. Trunk port không thuộc một mạng VLAN nào mà thuộc về tất cả các VLANs và do đó gửi từ  bất kỳ VLAN nào sẽ được chuyển qua trunk port để đến switch khác. Để có thể xác đinh VLAN ...
```
Phần này anh bổ sung thêm cách thứ 3 là nối 2 con switch với cùng một router.

thaonguyentien replied:
em viết thiếu ạ. Trunk port cũng dùng để nốt với router em sẽ bổ sung ạ

thaonguyentien committed:
Update VXLAN + Linux

Viet xong ve mot so khai niem va cau lenh trong linux

NTT-TNN commented:
@haminhcong @vanduc95 @huynhducbk95 @HaManhDong  review cho em với. Về phần linux em không biết có cần thêm gì không ạ?

thaonguyentien committed:
Update VXLAN + New Virtualization

Chinh sua hinh anh VXLAN
Bao cao ao hoa

hieulq reviewed (approved):
Anh approve PR này.
Review qua:
- Các khái niệm thì các em đã nắm được sơ lược, nhưng cách trình bày chưa thoáng lắm. Hiện tại cứ merge và 1 năm sau đọc lại xem thế nào nhé. ;)
- @haminhcong @huynhducbk95 @vanduc95 nên cho các em thực hành thêm để hiểu, đừng lý thuyết thuần quá.

NTT-TNN commented:
@hieulq  dạ vâng anh. Em cũng đang thử cài máy ảo và config mạng như anh @haminhcong giao trên trello nhưng em đang bị lỗi ở phần cài máy ảo bằng KVM ạ :)

Zalotuan0934346524 commented on K59/Nguyen-Tien-Thao/Linux/Linux_tutorial.md:
````diff
@@ -0,0 +1,359 @@
+
+```sh
+nguyentienthao@ntttnn-laptop:~$ ls
+build    distributted  official_bktranslator  Public           Videos
````
- - - [x] ![Uploading 200148264...]
